diff --git a/target/linux/rockchip/image/armv8.mk b/target/linux/rockchip/image/armv8.mk
index 62b68b9..b311abb 100644
--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -41,6 +41,15 @@ define Device/friendlyarm_nanopc-t6
 endef
 TARGET_DEVICES += friendlyarm_nanopc-t6
 
+define Device/friendlyarm_nanopi-m4b
+  DEVICE_VENDOR := FriendlyARM
+  DEVICE_MODEL := NanoPi M4B
+  SOC := rk3399
+  DEVICE_DTS := rockchip/rk3399-nanopi-m4b
+  DEVICE_PACKAGES := kmod-brcmfmac brcmfmac-firmware-43456-sdio
+endef
+TARGET_DEVICES += friendlyarm_nanopi-m4b
+
 define Device/friendlyarm_nanopi-r2c
   DEVICE_VENDOR := FriendlyARM
   DEVICE_MODEL := NanoPi R2C
