diff --git a/target/linux/rockchip/image/armv8.mk b/target/linux/rockchip/image/armv8.mk
--- a/target/linux/rockchip/image/armv8.mk
+++ b/target/linux/rockchip/image/armv8.mk
@@ -89,9 +89,17 @@ TARGET_DEVICES += friendlyarm_nanopc-t6
 TARGET_DEVICES += friendlyarm_nanopc-t6
 
 define Device/friendlyarm_nanopi-r2c
   $(Device/rk3328)
   DEVICE_VENDOR := FriendlyARM
   DEVICE_MODEL := NanoPi R2C
   DEVICE_PACKAGES := kmod-usb-net-rtl8152
 endef
+define Device/friendlyarm_nanopi-m4b
+  $(Device/rk3399)
+  DEVICE_VENDOR := FriendlyARM
+  DEVICE_MODEL := NanoPi M4B
+  DEVICE_DTS := rk3399-nanopi-m4b
+  DEVICE_PACKAGES := kmod-brcmfmac brcmfmac-firmware-43456-sdio
+endef
+TARGET_DEVICES += friendlyarm_nanopi-m4b
 # TARGET_DEVICES += friendlyarm_nanopi-r2c
